Remove emptyset from empty SimplicialComplex

gmou3 opened the pull request:
Allow a `SimplicialComplex` to be faceless, rather than always forcing it to include the `emptyset`.

```
sage: SimplicialComplex([])  # no faces
Simplicial complex with vertex set () and facets {}
sage: SimplicialComplex([()])  # one face
Simplicial complex with vertex set () and facets {()}
```
Previously the result of these commands was the same (with `facets {()}`). This change is advantageous in cases where a computation should result in the empty complex instead of the one with one face; an example of this is the broken circuit complex of a matroid with a loop.

gmou3 committed:
Remove emptyset face from empty SimplicialComplex

gmou3 committed:
Merge remote-tracking branch 'sagemath/develop' into empty_simplicial_complex

gmou3 committed:
Add broken_circuit_complex test

When there exists a loop, the empty set is a broken circuit, and should not be included in the broken circuit complex

fchapoton commented on src/sage/topology/cell_complex.py:
```diff
@@ -165,7 +165,7 @@ def dimension(self):
         try:
-            return max([x.dimension() for x in self._facets])
+            return max([-2] + [x.dimension() for x in self._facets])
         except AttributeError:
```
this could use `max(iterator, default=-2)`

gmou3 replied:
Thanks; that's better.

gmou3 committed:
Use "default" keyword in max

gmou3 committed:
Empty complex refinements

Set default simplicial complex dimension to -1, count empty cube in cubical complex

gmou3 committed:
Change representation of cubical complexes

Now of the form "Cubical complex with N vertices and M nonempty cubes", as suggested by jhpalmieri

gmou3 committed:
Merge branch 'develop' into empty_simplicial_complex

gmou3 committed:
Merge branch 'develop' into empty_simplicial_complex
